### PR TITLE
Fix init segfault where InitLoadWallet() calls ATMP before genesis

### PR DESCRIPTION
**Pull request by TheBlueMatt**
I honestly have no idea how the wallet.py tests are passing for anyone, but they reliably cause segfault here for me.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
utACK 37aefff5fcf7169a1b07ff8939850f630640f7e7

File: src/init.cpp [hunk 1]
```diff
@@ -1488,6 +1488,13 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         uiInterface.NotifyBlockTip.disconnect(BlockNotifyGenesisWait);
     }
 
+#ifdef ENABLE_WALLET
```
Comment (laanwj) [inline src/init.cpp]: It's unfortunate that this has to introduce a new `#ifdef ENABLE_WALLET` here, and ref to pWalletMain.
We're trying to get rid of them to remove the circular dependency between` _wallet` and `_server` libs (#7965), and in the interest of multi-wallet support.

**Comment by laanwj** [comment]
utACK otherwise

File: src/init.cpp (hunk 1 above)
Comment (MarcoFalke) [inline src/init.cpp]: It could be combined with the guard a few lines below, but this won't solve the larger problem.

I think for a bugfix this is fine.

**Review by jonasschnelli — CHANGES_REQUESTED** [review]

File: src/init.cpp (hunk 1 above)
Comment (laanwj) [inline src/init.cpp]: Yes I agree, let's merge this as it is a necessary bugfix, this can be improved later.

File: src/init.cpp [hunk 2]
```diff
@@ -1488,6 +1488,13 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
+    // Add wallet transactions that aren't already in a block to mempool
+    // Do this here as mempool requires genesis block to be loaded
+    if (pwalletMain)
+        pwalletMain->ReacceptWalletTransactions();
```
Comment (laanwj) [inline src/init.cpp]: @jonasschnelli we can do this later

**Pull request merged**